bitcoin/chainparams.c

```diff
@@ -48,6 +50,21 @@ const struct chainparams networks[] = {
      .p2sh_version = 196,
      .testnet = true,
      .bip32_key_version = {.bip32_pubkey_version = BIP32_VER_TEST_PUBLIC, .bip32_privkey_version = BIP32_VER_TEST_PRIVATE}},
+    {.network_name = "signet",
+     .bip173_name = "sb",
+     .bip70_name = "signet",
+     .genesis_blockhash = {{{.u.u8 = {0x2d, 0x09, 0x8b, 0x08, 0x55, 0x1b, 0xf3, 0xda, 0xb1, 0xa2, 0x72, 0xa8, 0x71, 0x4a, 0x12, 0x9d, 0x06, 0x5c, 0xfb, 0x32, 0xfa, 0x7e, 0x18, 0xdf, 0x00, 0x99, 0x09, 0x5f, 0xbc, 0xd8, 0x53, 0x62}}}},
+     .rpc_port = 38332,
+     .cli = "bitcoin-cli",
+     .cli_args = "-signet",
+     .dust_limit = { 546 },
+     .max_funding = AMOUNT_SAT_INIT((1 << 24) - 1),
+     .max_payment = AMOUNT_MSAT_INIT(0xFFFFFFFFULL),
+     .when_lightning_became_cool = 1,
+     .p2pkh_version = 125,
+     .p2sh_version = 87,
+     .testnet = false,
```

Comment:
I'd argue that we should set this to true. see `lightningd/bitcoin.c:380-387` and `lightningd/options.c:685--688` for rationale.